### ISSUE TITLE
Webhook event for COMMENT_ENTERED_MODERATION_QUEUE

### ISSUE THREAD

**Issue by ddrysdale-ffx** [coralproject/talk#4668]
As an engineer, I want to be notified when a comment is entered into PREMOD status, so that I can integrate with external platforms.

## Search Terms

External Moderation Phases
Webhook

## Suggestion

We are using Coral v8.5.3 and are integrating with an external 3rd party for comment moderation. Unfortunately the External Moderation Phases feature is not appropriate as the 3rd party require a consistent comment ID, but external moderation phases occur prior to the comment being created in Coral (ie no ID).

Our business requires that all comments are moderated - hence use the setting "Pre-moderate all comments", which means that newly created comments are created with PREMOD status.

Suggestion is to expose the COMMENT_ENTERED_MODERATION_QUEUE event as a webhook event. This will provide the ability to learn that a new comment has been added with PREMOD status.

A similar feature was released in Coral v8.6.3 to expose the COMMENT_LEFT_MODERATION_QUEUE webhook event - https://github.com/coralproject/talk/pull/4380. This prior PR will help the addition of the corresponding COMMENT_ENTERED_MODERATION_QUEUE event, with only minor code changes required.

We would like to push up a PR for this change and get it into the Coral mainline code.

## Use Cases

Integration with an external 3rd party comment moderation platform. They require a consistent comment ID as part of the data they are sent when the comment is first moderated and re-sent later to learn of the human decision in the case the comment required human moderation.

I found another prior issue from 2022 with the exact same problem, which was closed with no outcome - https://github.com/coralproject/talk/issues/3960.

The problem is two fold:
- the External Moderation Phases feature occurs BEFORE the comment is created, hence has no ID. The alternative is to use a webhook.
- the existing webhook events COMMENT_CREATED and COMMENT_REPLY_CREATED cannot be used as they are only sent when the comment is "published" (status is NONE or APPROVED). We need to know when a new comment is PREMOD.

Resulting webhook event data for COMMENT_ENTERED_MODERATION_QUEUE:

```
{
  id: 'd95d75ab-55d6-4d98-98a6-3f2980e7eaa5',
  createdAt: '2024-09-29T07:21:27.058Z',
  data: {
    queue: 'PENDING',
    commentID: 'c6ad8df2-b55d-4715-b4ef-68124e280f05',
    status: 'PREMOD',
    storyID: 'story-test',
    siteID: '373d38eb-b180-4772-b16f-85b14e87855b'
  },
  type: 'COMMENT_ENTERED_MODERATION_QUEUE',
  tenantID: '91d6e62e-ba7d-497d-8d0d-e4d56d48d33b',
  tenantDomain: 'localhost'
}
```

Note: multiple events are issued for a new comment such as this - PENDING and UNMODERATED queues. The webhook handler can determine which events to process based on queue and status.

**Comment by losowsky** [comment]
Thanks! We'll take a look.

**Comment by ddrysdale-ffx** [comment]
@losowsky - I have a branch with the required changes which I have tested. It is effectively 2 lines of code plus some changes to the WEBHOOKS.md doco (the prior change to add COMMENT_LEFT_MODERATION_QUEUE as a webhook did alot of the required work).

However, I am unable to push a branch to the repo. Failed permissions.

**Comment by losowsky** [comment]
Thanks! We’ll take a look soon.

**Comment by losowsky** [comment]
And if you fork the repo, you'll be able to open a pull request against the main repo, and we can take a look at your code. Thanks!

**Comment by ddrysdale-ffx** [comment]
Cool. Thanks for the info. Will do that now and post back here when the PR is up.

**Comment by ddrysdale-ffx** [comment]
Raised PR - https://github.com/coralproject/talk/pull/4673

**Comment by ddrysdale-ffx** [comment]
Thank you so much for getting this released so quickly. Really helped the work we are doing with external moderation

**Comment by ddrysdale-ffx** [comment]
Merged and released - v9.4.0
